oshmem/mca/spml/ucx/spml_ucx_component.c

```diff
@@ -491,8 +492,16 @@ static int mca_spml_ucx_component_fini(void)
 
 
     ret = opal_common_ucx_mca_pmix_fence_nb(&fenced);
-    if (OPAL_SUCCESS != ret) {
-        return ret;
+    switch (ret) {
+        case PMIX_SUCCESS:
+            break;
+
+        case PMIX_ERR_UNREACH:
+            /* It's ok if we are disconnected, continue cleanup */
+            fenced = 1;
+            break;
+
+        default: return ret;
```

Comment:
would it also make sense to set `fenced = 1` also for default case, maybe with a log?

Reply:
No, I don't think it makes sense.
`fenced` is just a local variable on stack used below to safely clean up the resources.
If we return an error, no need to set it

Reply:
i meant, setting it to 1 and continuing with cleanup anyways, wdyt?

Reply:
Ok, I see your point now.
Basically it means that if we don't manage to contact the server, we just go on with clean up.
To me it looks reasonable, because if we return an error here and interrupt the cleanup, then we remain in semi-destroyed state, and subsequent cleanup call will crash anyway.

On the other side, existing code in `_shmem_finalize` follows the pattern: check status of each cleanup stage and return an error:
https://github.com/open-mpi/ompi/blob/main/oshmem/runtime/oshmem_shmem_finalize.c#L90

Reply:
ok we could keep it like this then